app/assets/javascripts/cart.js

```diff
@@ -0,0 +1,171 @@
+function shoppingCart(cartName) {
+    this.cartName = cartName;
+    this.items = [];
+    this.clearCart = false;
+
+    this.loadItems();
+    var self = this;
+    $(window).unload(function () {
+        if (self.clearCart) {
+            self.clearItems();
+        }
+        self.saveItems();
+        self.clearCart = false;
+    });
+}
+
+shoppingCart.prototype.loadItems = function () {
+    var items = localStorage !== null ? localStorage[this.cartName + "_items"] : null;
+    if (items !== null && JSON !== null) {
+        try {
+            items = JSON.parse(items);
+            for (var i = 0; i < items.length; i++) {
+                var item = items[i];
+                if (item.code !== null && item.name !== null && item.price !==
+                    null && item.quantity !== null &&
+                        item.image !== null) {
+                    item = new CartItem(
```

Comment:
I think we need to have the `id` of the item, some we can easily recognize the item at the backend and get more information about the item